src/ui/pages/equipment/asset/AssetGeneral.js

```diff
@@ -7,6 +7,22 @@ import EAMAutocomplete from 'eam-components/dist/ui/components/inputs-ng/EAMAuto
 import EAMSelect from 'eam-components/dist/ui/components/inputs-ng/EAMSelect';
 import { isDepartmentReadOnly, isMultiOrg } from 'ui/pages/EntityTools';
 import EAMUDF from 'ui/components/userdefinedfields/EAMUDF';
+import GridWS from 'eam-components/dist/ui/components/eamgrid/lib/GridWS';
+
+async function fetchHasHazards(equipmentCode) {
+    const gridRequest = {
+        rowCount: 1,
+        params: {
+            //equipmentno: equipmentCode,
+            "parameter.object": equipmentCode,
+            "parameter.objorganization": "*",
+        },
+        gridName: "OSOBJA_ESF",
+        userFunctionName: 'OSOBJA'
+    };
+    const gridData = await GridWS.getGridData(gridRequest);
+    return gridData.body.data.records !== '0';
+}
```

Comment:
To keep it more flexible, I suggest modifying this method to retrieve the hazards instead of simply checking if hazards exist. Pass the list of hazards to the `StatusRow`, and let the `StatusRow` determine how to handle them. This approach allows for potential future changes, such as displaying different icons depending on the specific hazards.